buildinfo/build_info.go

```diff
@@ -0,0 +1,102 @@
+/*
+ * Copyright 2022 Aspect Build Systems, Inc.
+ *
+ * Licensed under the aspect.build Commercial License (the "License");
+ * you may not use this file except in compliance with the License.
+ * Full License text is in the LICENSE file included in the root of this repository.
+ */
+
+package buildinfo
+
+import (
+	"fmt"
+	"strings"
+)
+
+const (
+	// Git status
+	CleanGitStatus = "clean"
+
+	// Release values
+	PreStampRelease = "no release"
+
+	// Version constants
+	NotCleanVersionSuffix = " (with local changes)"
+	NoReleaseVersion      = "unknown [not built with --stamp]"
+)
+
+type VersionFormat int
+
+const (
+	// VersionFormat
+	ConventionalFormat VersionFormat = iota
+	GNUFormat
+)
+
+type BuildInfo struct {
+	BuildTime string
+	HostName  string
+	GitCommit string
+	GitStatus string
+	Release   string
+}
+
+func New(
+	buildTime string,
+	hostName string,
+	gitCommit string,
+	gitStatus string,
+	release string,
+) *BuildInfo {
+	return &BuildInfo{
+		BuildTime: buildTime,
+		HostName:  hostName,
+		GitCommit: gitCommit,
+		GitStatus: gitStatus,
+		Release:   release,
+	}
+}
+
+func Current() *BuildInfo {
+	return New(
+		BuildTime,
+		HostName,
+		GitCommit,
+		GitStatus,
+		Release,
+	)
+}
+
+func (bi BuildInfo) HasRelease() bool {
+	return bi.Release != "" && bi.Release != PreStampRelease
+}
+
+func (bi BuildInfo) IsClean() bool {
+	return bi.GitStatus == CleanGitStatus
+}
+
+func (bi BuildInfo) Version() string {
```

Comment:
I'm a little confused by naming, what does "Utility" mean - seems like this function is more like Release() and the one below is Version()

Reply:
There is already a field called `Release`. It represents the stamped release. The `Version()` function applies logic that decorates the release info. I went ahead and renamed `UtilityVersion` to `CommandVersion`. It outputs the command-line application's version which includes the name of the application along with the `Version()`.